src/interceptors/WebSocket/implementations/WebSocketClass/WebSocketClassInterceptor.ts

```diff
@@ -0,0 +1,266 @@
+import { invariant } from 'outvariant'
+import type { WebSocketEventsMap } from '../../index'
+import { Interceptor } from '../../../../Interceptor'
+import { WebSocketClassClient } from './WebSocketClassClient'
+import { WebSocketClassServer } from './WebSocketClassServer'
+import type { WebSocketSendData } from '../../WebSocketTransport'
+import { bindEvent } from '../../utils/bindEvent'
+import { WebSocketClassTransport } from './WebSocketClassTransport'
+
+export class WebSocketClassInterceptor extends Interceptor<WebSocketEventsMap> {
+  static symbol = Symbol('websocket-class')
+
+  constructor() {
+    super(WebSocketClassInterceptor.symbol)
+  }
+
+  protected checkEnvironment(): boolean {
+    // Enable this interceptor in any environment
+    // that has a global WebSocket API.
+    return typeof globalThis.WebSocket !== 'undefined'
```

Comment:
The WebSocket interceptor will be available anywhere where the global `WebSocket` class is present. 

We are setting that class in a custom Vitest environment, taking it from `undici` and making it a global class.